### ISSUE TITLE
Allow configurable variable name for "react-in-jsx-scope" rule

### ISSUE THREAD

**Issue by montemishkin** [yannickcr/eslint-plugin-react#228]
#### short explanation

It would be nice to be able to specify the name of the react import (for instance as `react` rather than `React`).

#### slightly longer explanation

With the current implementation, the "react-in-jsx-scope" rule can be set like so

```
"react-in-jsx-scope": 2
```

which causes the linting to throw an error if jsx is used without a variable named `React` being in scope.

It would be nice to additionally have the option to specify that the variable be called `react` instead, as this allows any given codebase using react to adhere to its own name-casing conventions.  The usage might look like

```
"react-in-jsx-scope": [2, "react"]
```

indicating that the linting should throw an error if jsx is used without a variable named `react` being in scope.

**Comment by yannickcr** [comment]
Hi,

How do you configure your compiler to compile to `react` instead of `React`, [Babel `jsxPragma` option](http://babeljs.io/docs/usage/options/) ?

Maybe we could rely on it to do the check. Right now we only support the `/** @jsx pragma */` instruction.

It could lead us to this (in order of priority):

1. Rule configuration in `.eslintrc`
2. Babel `jsxPragma` option (from `.babelrc`)
2. `/** @jsx pragma */` comment
2. `React` (default)

What do you think of it ?

**Comment by montemishkin** [comment]
Yes, I set Babel's jsxPragma option to "react.createElement".

Just in case it matters, I'll note that I use webpack to run my js files through the babel-loader, so the loader configuration ends up looking something like this

```
loaders: [
    {
        test: /\.js$/,
        loader: "babel?stage=0&jsxPragma=react.createElement",
    },
],
```

Your suggested order of priority looks good to me.

Thanks for looking into this!
